R/df2bib.R

```diff
@@ -24,7 +24,7 @@ df2bib <- function(x, file = "", append = FALSE) {
   if (!is.character(file)) {
     stop("Invalid file path: Non-character supplied.", call. = FALSE)
   }
-  if (as.numeric(file.access(dirname(file), mode = 2)) != 0 && file != "") {
+  if (as.numeric(file.access(dirname(file), mode = 2))[1] != 0 & file != "") {
```

Comment:
Personally, I'd prefer a comment regarding the use `[1]` here or dropping it altogether:

```suggestion
  if (as.numeric(file.access(dirname(file), mode = 2)) != 0 & file != "") {
```

IMHO the variable name `file` strongly implies the invariant `length(file) == 1`. (But maye you'd prefer an explicit check above after the `is.character(file)` one?). Thus, your explicit use of `[1]` rather adds confusion for me regarding why `dirname()` or `file.access()` would return more values than passed in.

Reply:
Hi @mschilli87 ! Thanks for the quick reply and your detailed suggestions! I completely agree with the changes you're suggesting. My aim was to stick as close to the previous code as possible as I did not know whether there were any edge cases that might arise as a result (e.g. where `length(file) > 1`). I also don't see a case where this would be the case so your solution works for me!